main.go

```diff
@@ -133,6 +134,46 @@ func (d *Downloader) downloadFileWithTimeout(userCtx context.Context, u string)
 	}
 }
 
+func (d *Downloader) getSize(ctx context.Context, u string) (uint64, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodHead, u, nil)
+	if err != nil {
+		return 0, fmt.Errorf("creating the request for %s: %w", u, err)
+	}
+	resp, err := d.client.Do(req)
+	if err != nil {
+		return 0, fmt.Errorf("sending get http request to %s: %w", u, err)
+	}
+	if resp.StatusCode != 200 {
+		return 0, fmt.Errorf("got unexpected http response status for %s: %s", u, resp.Status)
+	}
+	defer resp.Body.Close()
```

Comment:
We need to close the body even if the status is not what we expect.

```suggestion
	defer resp.Body.Close()
	if resp.StatusCode != 200 {
		return 0, fmt.Errorf("got unexpected http response status for %s: %s", u, resp.Status)
	}
```